### ISSUE TITLE
No warning or suggestion when non-nullable reference is compared against null

### ISSUE THREAD

**Issue by wanton7** [dotnet/roslyn#38871]
**Version Used**: 
Microsoft Visual Studio 2019 Community Version 16.3.1
.NET Core Version 3.0.100

In csproj we have this
```xml
<PropertyGroup>
    <TargetFramework>netcoreapp3.0</TargetFramework>
    <LangVersion>8.0</LangVersion>
    <Nullable>enable</Nullable>
</PropertyGroup>
```

**Steps to Reproduce**:

1. 
```csharp
public class Foo
{
  public string Name { get; set; } = string.Empty;
}
```
2. 
```csharp
var foo = new Foo();
if (foo.Name == null)
{
  // stuff
}
```

**Expected Behavior**:
`if (foo.Name == null)` would produce warning or suggestion. I would prefer warning.

**Actual Behavior**:
No warning or suggestion from `if (foo.Name == null)`

**Comment by CyrusNajmabadi** [comment]
This is expected behavior.  NullableReferenceTypes are a best-effort analysis that may be incorrect at runtime.  Code still needs actual null checks to prevent from errant nulls slipping in.  As such, the compiler does not warn here as it may be absolutely necessary to have these null checks for proper behaving code.

**Comment by wanton7** [comment]
Then could you this be added as option so I could turn on warnings or suggestions for it with .editorconfig? We switched to nullable reference types and properties where accidentally left to non-nullable. It would been really helpful us if we had noticed issue like this during compilation. Actual class was deserialized from JSON so all properties it had should have been non-nullable to begin with.

**Comment by jcouv** [comment]
This behavior was an explicit design decision. We let users code defensively. This is important for scenarios like implementing public APIs: even if the API is declared as non-nullable, the code should check and handle `null` values that callers might pass.

That said, I could imagine an analyzer that would offer a suggestion when a comparison seems un-necessary.
